src/HTTP/UrlClient.cpp

```diff
@@ -493,15 +493,17 @@ void cUrlClientRequest::RedirectTo(const AString & a_RedirectUrl)
 		return;
 	}
 
+	// Keep ourself alive while the link drops ownership
+	auto Self = m_Self.lock();
+
 	// Do the actual redirect:
-	m_Link->Close();
+	m_Link.lock()->Close();
```

Comment:
`lock()` could return an emtpy `shared_ptr`. And then everything will go 💥 .
So it should be:

    if(auto link = m_Link.lock())
        link->Close();